### DISCUSSION TITLE
torch.compile TensorProduct

### DISCUSSION THREAD

**Discussion by mario-linov** [e3nn/e3nn#431] [Q&A] [answered]
Hi all,

Has anyone tried successfully to torch.compile the TensorProduct module in [e3nn/o3/_tensor_product/_tensor_product.py](e3nn/o3/_tensor_product/_tensor_product.py)? I have been trying it, but found some issues. 

First, it is needed to torch.compiler.disable the function `change_basis_real_to_complex` in [e3nn/o3/_wigner.py](e3nn/o3/_wigner.py). That was not a big problem.

After that I get the error: 

```
     check_fn = CheckFunctionManager(
               ^^^^^^^^^^^^^^^^^^^^^
  File "/home/lino/miniconda3/envs/gfd/lib/python3.11/site-packages/torch/_dynamo/guards.py", line 930, in __init__
    self.check_fn = self.compile_check_fn(
                    ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lino/miniconda3/envs/gfd/lib/python3.11/site-packages/torch/_dynamo/guards.py", line 989, in compile_check_fn
    add_code_part(code, gcl.guard)
  File "/home/lino/miniconda3/envs/gfd/lib/python3.11/site-packages/torch/_dynamo/guards.py", line 970, in add_code_part
    maybe_stack = f"\nStack:\n{''.join(guard.stack.format())}"
                                       ^^^^^^^^^^^^^^^^^^
torch._dynamo.exc.InternalTorchDynamoError: 'NoneType' object has no attribute 'format'
```

Any clues?

**Comment by mitkotak** [comment]
Is there a MWE (minimal working example) that I can use to replicate your error ?

**Comment by mitkotak** [comment, marked as answer]
@mario-linov Sorry took a while but works with this https://github.com/e3nn/e3nn/pull/455
